addons/l10n_latam_invoice_document/models/account_move.py

```diff
@@ -141,8 +141,8 @@ def _post(self, soft=True):
     @api.constrains('name', 'journal_id', 'state')
     def _check_unique_sequence_number(self):
         """ This uniqueness verification is only valid for customer invoices, and vendor bills that does not use
-        documents. A new constraint method _check_unique_vendor_number has been created just for validate for this purpose """
-        vendor = self.filtered(lambda x: x.is_purchase_document() and x.l10n_latam_use_documents)
+        documents. A new constraint method _check_unique_number_by_partner has been created just for validate for this purpose """
+        vendor = self.filtered(lambda x: x._should_check_unique_number_by_partner() and x.l10n_latam_use_documents)
```

Comment:
I suggest we even do this code:

`    def _is_manual_document_number(self):
        return self._should_check_unique_number_by_partner()`

Reply:
This suggestion has been implemented in this PR, I think it improves latam document type module a bit. Please review it.